caddyhttp/staticfiles/fileserver.go

```diff
@@ -121,13 +125,36 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request, name stri
 		return http.StatusNotFound, nil
 	}
 
+	filename := d.Name()
+
+encodings:
+	for _, encoding := range staticEncodingPriority {
+		if strings.Contains(r.Header.Get("Accept-Encoding"), encoding) {
+			gf, err := fs.Root.Open(location + staticEncoding[encoding])
+			if err == nil {
+				defer gf.Close()
+
+				stat, err := gf.Stat()
+				if err == nil {
+					d = stat
```

Comment:
It don't need to. Filename is only needed so the ```http.ServeContent``` will return valid mime-type - it has to point to original filename, not compressed one :)
So: ```http.ServeContent(w, r, "index.html", compressedIndex.ModTime(), compressedIndexFile)``` is intended :)